admin_guide/custom_controllers.adoc

```diff
@@ -0,0 +1,300 @@
+[[admin-guide-custom-controllers]]
+= Creating Custom Controllers
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+
+toc::[]
+
+
+
+In the Kubernetes API, a controller is a control loop that watches the shared state of a cluster 
+through the API server and makes changes in an attempt to move the current state towards the desired state. 
+Examples of controllers that ship with {product-title} today are 
+xref:../architecture/core_concepts/deployments.adoc#replication-controllers[replication controllers] and 
+xref:../dev_guide/application_health.adoc#container-health-checks-using-probes[endpoints controllers].
```

Comment:
What's the meaning of this link? It points to health checks of apps. I don't see a controller their, especially no endpoints controller.

Reply:
Assuming https://docs.openshift.com/container-platform/3.7/dev_guide/application_health.html is what this will point to....

Reply:
@sttts Can you suggest appropriate references? Or, do folks know what controllers are?

Reply:
- DaemonSet (makes sure that a pod is running on each node)
- the scheduler (makes sure that every pod is assigned to a node)
- the endpoint controller (makes sure that for every pod belong to a service, its IP address is in the endpoints of the service)
- the physical volume controller (makes sure that for every PhysicalVolumeClaim there is an actual volume created with the cloud provider).